code/ui/components/src/components/icon/icon.tsx

```diff
@@ -24,18 +24,27 @@ export interface IconsProps extends ComponentProps<typeof Svg> {
   icon: IconType;
   useSymbol?: boolean;
   onClick?: () => void;
+  removeDeprecateWarning?: boolean;
 }
 
 /**
  * @deprecated No longer used, will be removed in Storybook 9.0
  * Please use the `@storybook/icons` package instead.
  * */
-export const Icons = ({ icon, useSymbol, ...props }: IconsProps) => {
-  deprecate(
-    `Use of the deprecated Icons ${
-      `(${icon})` || ''
-    } component detected. Please use the @storybook/icons component directly. For more informations, see the migration notes at https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#icons-is-deprecated`
-  );
+export const Icons = ({
+  icon,
+  useSymbol,
+  removeDeprecateWarning = false,
```

Comment:
I'd suggest `__suppressDeprecationWarning` instead.

1. the `__` prefix highlights that this is a prop only intended for internal use
2. `remove` is a destructive action, we're not really removing it everywhere, we're just suppressing it here.

Reply:
Done